diff --git a/libretro.cpp b/libretro.cpp
--- a/libretro.cpp
+++ b/libretro.cpp
@@ -391,13 +391,4 @@ void retro_deinit(void)
 {
    if (surf)
-   {
-#if defined(WANT_32BPP)
-      if (surf->pixels)
-         free(surf->pixels);
-#elif defined(WANT_16BPP)
-      if (surf->pixels16)
-         free(surf->pixels16);
-#endif
       delete surf;
-   }
    surf = NULL;
@@ -518,4 +509,6 @@ void *retro_get_memory_data(unsigned type)
 {
    if ( type == RETRO_MEMORY_SYSTEM_RAM )
       return SNES::memory::wram.data();
+   if ( type == RETRO_MEMORY_SAVE_RAM )
+      return SNES::memory::cartram.data();
@@ -524,4 +517,6 @@ size_t retro_get_memory_size(unsigned type)
 {
    if ( type == RETRO_MEMORY_SYSTEM_RAM )
       return SNES::memory::wram.size();
+   if ( type == RETRO_MEMORY_SAVE_RAM )
+      return SNES::memory::cartram.size();
