diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,26 +9,8 @@
 project(WLA-DX C)
 
 set(ARCHS
     # wla-NAME:compile-def
-    "gb:GB"
-    "cx4:CX4"
-    "65c02:WDC65C02"
-    "65ce02:CSG65CE02"
-    "6502:MCS6502"
     "65816:W65816"
-    "huc6280:HUC6280"
-    "spc700:SPC700"
-    "z80:Z80"
-    "z80n:Z80N"
-    "ez80:EZ80"
-    "68000:MC68000"
-    "6800:MC6800"
-    "6801:MC6801"
-    "6809:MC6809"
-    "8008:I8008"
-    "8080:I8080"
-    "superfx:SUPERFX"
-    "sh2:SH2"
     )
 
 option(GDB_DEBUGGING
@@ -119,8 +101,5 @@
 include_directories("${PROJECT_SOURCE_DIR}")
 
-add_subdirectory(wlab)
 add_subdirectory(wlalink)
-add_subdirectory(byte_tester)
-add_subdirectory(doc)
 
 add_custom_target(generators)
